Framework/Indexing/src/IndexInfo.cpp

```diff
@@ -320,7 +322,7 @@ IndexInfo::globalSpectrumIndicesFromDetectorIndices(
     spectrumIndices.resize(*status.count<int64_t>());
   }
 
-  if (detectorIndices.size() != spectrumIndices.size())
+  if (detectorIndices.size() > spectrumIndices.size())
```

Comment:
Is this check sufficient? From our discussion on Slack:

"It is actually more complicated: You need to ensure that all requested detectors have a matching spectrum. Now, if you allow mapping 1 detector to multiple spectra, how can you catch cases with a detector that has 0 spectra? I think you need to add some extra code for this.
Say you have 2 detectors, and the output is 2 spectra. If it is det1->spec1 and det2->spec2 it is ok, but if it is det1 -> spec1, spec2 and det1->none it must fail."

Reply:
It is exactly for this case, that I added the throw at L311, when there is a c==1.